packages/umbra_flutter/lib/src/umbra_exception.dart

```diff
@@ -0,0 +1,215 @@
+/// {@template umbra_exception}
+/// Exception thrown when an error occurs during shader compilation.
+/// {@endtemplate}
+abstract class UmbraException extends Error {
+  /// {@macro umbra_exception}
+  factory UmbraException(Exception exception) {
```

Comment:
I'm confused...this is a factory for `UmbraException` but it never returns an `UmbraException`?

Reply:
It returns a `UnsupportedOperator` or if it cant determine what kind of transpile exception it is it throws an exception related to that. 

Now that I think about it, instead of doing that I should just rethrow the original one.

Reply:
Ah oops I misread and thought we were throwing no matter what. This makes more sense 👍